controllers/machinedeletionremediation_controller.go

```diff
@@ -217,9 +249,11 @@ func (r *MachineDeletionRemediationReconciler) getMachine(remediation *v1alpha1.
 			if gotMachineFromNode {
 				// The Machine's Name and Ns came from the related Node, it was not expected not to find the Machine
 				r.Log.Error(err, noMachineFoundError, "node", remediation.Name, "machine", machineName)
+				r.updateConditions(remediationFailed, remediation)
```

Comment:
I don't think `updateConditions` should be ignored, or at least explained why it's ignored